cmd/subnetcmd/deploy.go

```diff
@@ -187,62 +184,82 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 	return deploySubnet(cmd, args)
 }
 
-func createMainnetGenesis(chain string) error {
-	evmGenesis, err := app.LoadEvmGenesis(chain)
-	if err != nil {
-		return err
+func updateSubnetEVMGenesisChainID(genesisBytes []byte, newChainID uint) ([]byte, error) {
+	var genesisMap map[string]interface{}
+	if err := json.Unmarshal(genesisBytes, &genesisMap); err != nil {
+		return nil, err
 	}
-	ux.Logger.PrintToUser("Enter your subnet's ChainID. It can be any positive integer.")
-	var chainID *big.Int
-	if mainnetChainID != "" {
-		newChainID := new(big.Int)
-		newChainID, ok := newChainID.SetString(mainnetChainID, 10)
-		if !ok {
-			return errors.New("SetString: error")
-		}
-		chainID = newChainID
-	} else {
-		chainID, err = app.Prompt.CapturePositiveBigInt("ChainID")
-		if err != nil {
-			return err
-		}
+	configI, ok := genesisMap["config"]
+	if !ok {
+		return nil, fmt.Errorf("config field not found on genesis")
 	}
-	evmGenesis.Config.ChainID = chainID
-	jsonBytes, err := evmGenesis.MarshalJSON()
-	if err != nil {
-		return err
+	config, ok := configI.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("expected genesis config field to be a map[string]interface, found %T", configI)
 	}
-	var prettyJSON bytes.Buffer
-	err = json.Indent(&prettyJSON, jsonBytes, "", "    ")
+	config["chainId"] = float64(newChainID)
+	return json.MarshalIndent(genesisMap, "", "  ")
+}
+
+// updates sidecar with genesis mainnet id to use
+// given either by cmdline flag, original genesis id, or id obtained from the user
+func getSubnetEVMMainnetChainID(sc *models.Sidecar, subnetName string) error {
+	// get original chain id
+	evmGenesis, err := app.LoadEvmGenesis(subnetName)
 	if err != nil {
 		return err
 	}
-	return app.WriteGenesisMainnetFile(chain, prettyJSON.Bytes())
-}
-
-func handleMainnetChainID(chain string) error {
-	genesisMainnetPath := app.GetGenesisMainnetPath(chain)
-	_, err := os.ReadFile(genesisMainnetPath)
-	if err != nil && os.IsNotExist(err) {
+	if evmGenesis.Config == nil {
```

Comment:
nit: can be combined into `if evmGenesis.Config == nil || evmGenesis.Config.ChainID == nil`

Reply:
It should be &&. But I prefer to keep separate error msgs.